### PR TITLE
[FLINK-13277][hive] add documentation of Hive source/sink

### PR DESCRIPTION
**Pull request by lirui-apache**
## What is the purpose of the change

To add documentation for hive connector.

## Brief change log

  - Added doc for hive connector.

## Verifying this change

Doesn't need testing.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? NA

### CODE REVIEW

**Comment by lirui-apache** [comment]
Hive table source and sink are not mentioned in the doc since they're not meant to be public APIs.
@xuefuz @zjuwangg please let me know if you find anything to add.

**Comment by xuefuz** [comment]
@lirui-apache Could you restart CI test? Previous failure doesn't seem related.

File: docs/dev/batch/connectors.md [hunk 1]
```diff
@@ -217,4 +217,21 @@ The example shows how to access an Azure table and turn data into Flink's `DataS
...
+
+### Limitations
+
+The following is a list of major limitations of the Hive connector.
+
+1. INSERT OVERWRITE is not supported.
+2. Inserting into partitioned tables is not supported.
+3. ACID tables are not supported.
+4. Bucketed tables are not supported.
+5. UNION type is not supported.
```
Comment (bowenli86) [inline docs/dev/batch/connectors.md]: we can move the type related limitations to https://github.com/apache/flink/blob/master/docs/dev/table/hive/index.md

File: docs/dev/batch/connectors.md [hunk 2]
```diff
@@ -217,4 +217,21 @@ The example shows how to access an Azure table and turn data into Flink's `DataS
 
 This [GitHub repository documents how to use MongoDB with Apache Flink (starting from 0.7-incubating)](https://github.com/okkam-it/flink-mongodb-test).
 
+## Hive Connector
```
Comment (bowenli86) [inline docs/dev/batch/connectors.md]: maybe there should be an additional cross page link with https://github.com/apache/flink/blob/master/docs/dev/table/hive/read_write_hive.md except that with catalog.md?

File: docs/dev/batch/connectors.md [hunk 3]
```diff
@@ -217,4 +217,21 @@ The example shows how to access an Azure table and turn data into Flink's `DataS
...
+
+Starting from 1.9.0, Apache Flink provides Hive connector to access Apache Hive tables. [HiveCatalog]({{ site.baseurl }}/dev/table/catalog.html#hivecatalog) is required in order to use the Hive connector.
+After HiveCatalog is setup, you can use Table API or Flink SQL Client to access tables in Apache Hive.
+Same as HiveCatalog, the officially supported Apache Hive versions are 2.3.4 and 1.2.1.
+
+### Limitations
```
Comment (bowenli86) [inline docs/dev/batch/connectors.md]: need to consolidate with https://github.com/apache/flink/blob/master/docs/dev/table/hive/read_write_hive.md#limitations

**Comment by lirui-apache** [comment]
Given that we already have `read_write_hive.md`, maybe we don't need a section for Hive connector in `connectors.md`. What do you guys think?

**Comment by xuefuz** [comment]
I think we can consolidate the content in the PR to the relevant content in read_write_hive.md and put a cross reference in connectors.md.

**Comment by lirui-apache** [comment]
@bowenli86 @xuefuz I have updated the PR to address your comments. And since we have decided not to support timestamp in 1.9.0, I also added that to the limitations part. Please have a look. Thanks.

File: docs/dev/table/hive/read_write_hive.md [hunk 4]
```diff
@@ -120,4 +120,11 @@ Flink SQL> INSERT INTO mytable (name, value) VALUES ('Tom', 4.72);
...
 
-Currently Flink's Hive data connector does not support writing into partitions. This feature is under active development.
+The following is a list of major limitations of the Hive connector. And we're actively working to close these gaps.
+
+1. INSERT OVERWRITE is not supported.
+2. Inserting into partitioned tables is not supported.
+3. ACID tables are not supported.
+4. Bucketed tables are not supported.
+5. Some data types are not supported. See the [limitations]({{ site.baseurl }}/dev/table/hive/#limitations) for details.
+6. Only a limited number of table storage formats have been tested, namely text, ORC, and Parquet.
```
Comment (xuefuz) [inline docs/dev/table/hive/read_write_hive.md]: Is squencefile tested?

File: docs/dev/batch/connectors.md [hunk 5]
```diff
@@ -217,4 +217,10 @@ The example shows how to access an Azure table and turn data into Flink's `DataS
 
 This [GitHub repository documents how to use MongoDB with Apache Flink (starting from 0.7-incubating)](https://github.com/okkam-it/flink-mongodb-test).
 
+## Hive Connector
+
+Starting from 1.9.0, Apache Flink provides Hive connector to access Apache Hive tables. [HiveCatalog]({{ site.baseurl }}/dev/table/catalogs.html#hivecatalog) is required in order to use the Hive connector.
+After HiveCatalog is setup, please refer to [Reading & Writing Hive Tables]({{ site.baseurl }}/dev/table/hive/read_write_hive.html) for how to use the Hive connector and its limitations.
```
Comment (xuefuz) [inline docs/dev/batch/connectors.md]: Nit: Reword it like "for the usage of the Hive connector and its limitations" to be more clear.

File: docs/dev/table/hive/index.md [hunk 6]
```diff
@@ -263,3 +259,5 @@ The following limitations in Hive's data types impact the mapping between Flink
 * Flink's `TIMESTAMP_WITH_TIME_ZONE` is not supported by Hive
 * Flink's `TIMESTAMP_WITH_LOCAL_TIME_ZONE` is not supported by Hive
 * Flink's `MULTISET` is not supported by Hive
+* Hive's `UNION` type is not supported
```
Comment (bowenli86) [inline docs/dev/table/hive/index.md]: I'll make these couple changes of data types in a different PR so this PR can focus on Hive source/sink. Thanks!

File: docs/dev/table/hive/read_write_hive.md [hunk 7]
```diff
@@ -120,4 +120,11 @@ Flink SQL> INSERT INTO mytable (name, value) VALUES ('Tom', 4.72);
 
 ### Limitations
```
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]: one more limitations: cannot read Hive views due to limitations of planner

File: docs/dev/table/hive/read_write_hive.md (hunk 4 above)
Comment (bowenli86) [inline docs/dev/table/hive/read_write_hive.md]: do we need extra dependencies if users use ORC/Parquet?

A user trying to use lzo found that he needed to include hadoop-lzo himself
http://apache-flink.147419.n8.nabble.com/flink1-10-hive-td336.html

Comment (lirui-apache) [reply docs/dev/table/hive/read_write_hive.md]: I don't think so according to our previous tests. Also manually verified reading ORC/Parquet tables with different compression codec and seems we're good.

**Comment by lirui-apache** [comment]
@xuefuz @bowenli86 PR updated according to your feedbacks.

**Comment by bowenli86** [comment]
btw, we need to copy the exact same content from xxx.md to xxx.zh.md, to keep English and Chinese docs in sync and facilitate translation in the future

**Comment by lirui-apache** [comment]
@bowenli86 Thanks for letting me know. I have synced the changes to zh docs.

**Comment by bowenli86** [comment]
LGTM, merging

**Pull request closed without merging**